CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+<!--
+Copyright 2017 F5 Networks Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Contributing Guide for k8s-bigip-ctlr
+If you have found this that means you you want to help us out. Thanks in advance for lending a hand! This guide should get you up and running quickly and make it easy for you to contribute.  
+
+## Issues
+Creating issues is good, creating good issues is even better. Filing meaningful bug reports with lots of information in them helps us figure out what to fix when and how it impacts our users. We like bugs because it means people are using our code, and we like fixing them even more.
+
+Please follow these guidelines for filing issues.
+* Describe the problem
+* Include version of k8s, k8s-bigip-ctlr, and BIG-IP
+* Include detailed information about how to recreate the issue
+* Include relevant and sanitized (no identification) configuration, error messages and logs
+
+## Pull Requests
+The hooks that are automatically run as part of each pull request are:
+* Unit tests are executed
+* Code coverage data is collected
+* Formatting and linting checks are performed
+* Third party attributions are generated and checked
+* Product documentation is built
+* Container is built and pushed to dockerhub
+
+If you are submitting a pull request, you need to make sure that you have done a few things first.
+
+* If an issue doesn't exist, file one
+* Make sure you have tested your code
+* The master branch must be kept release ready at all times. This requires that a single commit should contain the code changes, unit tests, and documentation changes (if any)
+* Use proper formatting for the code
+* Clean up your git history because no one wants to see 75 commits for one issue
+*  Use the commit message format shown below
+
+## Commit Message Format
+The commit message for your final commit should use the following format:
+```
+<One line summarizing what changed>
+
+Problem: Brief description of the problem.
+
+Solution: Detailed description of your solution
+
+Testing (optional if not described in Solution section): Description of tests that were run to exercise the solutions (unit tests, system tests, etc)
+
+Fixes#<issue-number> (Or Fixes: #<issue number> if this commit fixes it)
```

Comment:
If the commit is fixing one or two bugs, it's best to put this in the title.

For instance:

```
Fix #31: Don't frob the foo before bar

Problem: We get segfaults when we frobbed foo without already bar.

Solution: Check if we need to bar before frobbing foo.

affects-branches: 1.1-stable
```

is better than:

```
Don't frob the foo before bar

Problem: We get segfaults when we frobbed foo without already bar.

Solution: Check if we need to bar before frobbing foo.

Fixes#31

affects-branches: 1.1-stable
```

because when I do "git history --pretty=oneline" or gitk or look at the headlines via github commits, I can instantly tell that's a bugfix.